In [1]:
import pandas as pd
import plotly.express as px

In [7]:
df = pd.read_csv("Compute_allocations_transformer.csv")
# byte_level = df[df['Patch size'] == 1]
byte_level = df[(df['FLOPs'] == 's') & (df['Patch size'] == 4)]
# byte_level['Global transformer parameters'].sort_values(ascending=False)
fig = px.histogram(byte_level['Global model parameters'])

fig.show()

In [13]:
byte_level[(byte_level['Global model parameters'] <= 1.3e8) & (byte_level['Global model parameters'] >= 1.1e8) & (byte_level["Global model layers"] >= 16)]

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
4327,s,4,9.963997e+09,2448424.96,1.199728e+08,9793699.84,896,512,16,4
4335,s,4,9.220995e+09,3098787.84,1.199728e+08,12395151.36,896,576,16,4
4470,s,4,1.008888e+10,2448424.96,1.163767e+08,9793699.84,832,512,18,4
4474,s,4,9.327847e+09,3098787.84,1.163767e+08,12395151.36,832,576,18,4
4549,s,4,9.012032e+09,3098787.84,1.228421e+08,12395151.36,832,576,19,4
4609,s,4,1.037069e+10,2448424.96,1.101791e+08,9793699.84,768,512,20,4
4613,s,4,9.568237e+09,3098787.84,1.101791e+08,12395151.36,768,576,20,4
4943,s,4,9.375604e+09,2448424.96,1.199728e+08,12242124.80,896,512,16,5
5061,s,4,1.032844e+10,1874575.36,1.163767e+08,9372876.80,832,448,18,5
5065,s,4,9.486091e+09,2448424.96,1.163767e+08,12242124.80,832,512,18,5


4e18 FLOPs
===

tokens, global model params, decoder params, global model dimension, encoder/decoder dimension, global model layers, decoder layers

Patch size = 2

4.094315e+10, 11247452,	1224212, 448, 256, 6, 2 (10M run)

1.711949e+10, 30987878,	2869248, 576, 320, 10, 3 (30M run)

9.492046e+09, 60177695, 5508956, 704, 384, 13, 4 (60M run)

Patch size = 4

4.937337e+10, 21691515, 1836319, 576, 256, 7, 3

1.847138e+10, 60177695, 5508956, 704, 384, 13, 4

9.375604e+09, 120M, 12242125, 896, 512, 16, 5